Kernel/Storage/StorageManagement.cpp

```diff
@@ -193,6 +193,136 @@ UNMAP_AFTER_INIT void StorageManagement::enumerate_disk_partitions()
+    metadata.size = file_size;
+    auto mode = TRY(convert_to_uint_from_inode_field(StringView { header.mode, sizeof(InitRAMFSInodeHeader::mode) }));
+    metadata.mode = mode;
+
+    metadata.uid = TRY(convert_to_uint_from_inode_field(StringView { header.uid, sizeof(InitRAMFSInodeHeader::uid) }));
+    metadata.gid = TRY(convert_to_uint_from_inode_field(StringView { header.gid, sizeof(InitRAMFSInodeHeader::gid) }));
+    metadata.link_count = TRY(convert_to_uint_from_inode_field(StringView { header.nlink, sizeof(InitRAMFSInodeHeader::nlink) }));
+
+    auto mtime = TRY(convert_to_uint_from_inode_field(StringView { header.mtime, sizeof(InitRAMFSInodeHeader::mtime) }));
+    metadata.atime = Time::from_timespec({ mtime, 0 });
+    metadata.ctime = Time::from_timespec({ mtime, 0 });
+    metadata.mtime = Time::from_timespec({ mtime, 0 });
+    metadata.dtime = Time::from_timespec({ mtime, 0 });
+    metadata.block_size = PAGE_SIZE;
+    // FIXME: Is this correct in CPIO archives?
+    metadata.block_count = static_cast<blkcnt_t>(static_cast<u64>(metadata.size) / static_cast<u64>(PAGE_SIZE));
+
+    if (Kernel::is_character_device(mode) || Kernel::is_block_device(mode)) {
+        auto dev = TRY(convert_to_uint_from_inode_field(StringView { header.rdev, sizeof(header.rdev) }));
+        metadata.major_device = (dev & 0xfff00) >> 8;
+        metadata.minor_device = (dev & 0xff) | ((dev >> 12) & 0xfff00);
+    }
+    if (metadata.is_directory()) {
+        return TmpFSInode::try_create_as_directory({}, fs, metadata, parent_directory_inode);
+    }
+
+    if (file_size == 0) {
+        return TmpFSInode::try_create_with_empty_content({}, fs, metadata, parent_directory_inode);
+    }
+
+    auto mapping_size = TRY(Memory::page_round_up(sizeof(InitRAMFSInodeHeader) + filename_length + file_size));
+    auto mapping = TRY(Memory::map_typed<u8>(current_address, mapping_size));
+    auto mapped_buffer = Span<u8> { mapping.base_address().offset(sizeof(InitRAMFSInodeHeader) + filename_length).as_ptr(), file_size };
+    return TmpFSInode::try_create_with_content({}, fs, metadata, mapped_buffer, parent_directory_inode);
+}
+
+ErrorOr<NonnullLockRefPtr<TmpFSInode>> StorageManagement::ensure_initramfs_path(TmpFSInode& inode, StringView full_name)
+{
+    auto first_path_part = full_name.find_first_split_view('/');
+    if (first_path_part == full_name)
+        return inode;
+    auto result = inode.lookup(first_path_part);
+    if (result.is_error()) {
+        VERIFY(result.error() != Error::from_errno(ENOENT));
+        return result.release_error();
+    }
+    auto next_inode = static_ptr_cast<TmpFSInode>(result.release_value());
+    return ensure_initramfs_path(*next_inode, full_name.substring_view(1 + first_path_part.length()));
+}
+
+UNMAP_AFTER_INIT ErrorOr<void> StorageManagement::populate_initramfs(TmpFS& fs, PhysicalAddress initramfs_image_start, PhysicalAddress initramfs_image_end)
```

Comment:
this initramfs initialization feels like it should be the responsibility of some kind of `InitRamFS` code, rather than putting it under storage management? Even if it's just a helper class that this storage managment class delegates to directly for the time being, I think these methods should be "packaged" differently.